### PR TITLE
Improve workflow when PGDATA is not empty during bootstrap

### PR DESCRIPTION
**Pull request by CyberDem0n**
Recently it has happened two times when people tried to deploy the new cluster but postgres data directory wasn't empty and also wasn't valid. In this case Patroni was still creating initialize key in DCS and trying to start the postgres up.
Now it will complain about non-empty invalid postgres data directory and exit.

Close https://github.com/zalando/patroni/issues/1216

### CODE REVIEW

File: patroni/postgresql/__init__.py [hunk 1]
```diff
@@ -200,7 +200,7 @@ def set_pending_restart(self, value):
 
     @property
     def sysid(self):
-        if not self._sysid and not self.bootstrapping:
+        if not self._sysid and not self.bootstrapping or not self.is_running():
```
Comment (sdudoladov) [inline patroni/postgresql/__init__.py]: Against which situation does the `or` clause protect ?

Comment (CyberDem0n) [reply patroni/postgresql/__init__.py]: I'll look at it one more time tomorrow morning.

File: tests/test_ha.py [hunk 2]
```diff
@@ -1027,7 +1027,6 @@ def test_update_cluster_history(self):
             self.assertEqual(self.ha.run_cycle(), 'no action.  i am the leader with the lock')
 
     @patch('sys.exit', return_value=1)
-    @patch('requests.get', requests_get)
```
Comment (sdudoladov) [inline tests/test_ha.py]: is it an unrelated change that simply removes a no longer needed mock ?

File: patroni/postgresql/__init__.py [hunk 3]
```diff
@@ -255,7 +255,8 @@ def query(self, sql, *args, **kwargs):
             raise PostgresConnectionException(str(e))
 
     def data_directory_empty(self):
-        return not os.path.exists(self._data_dir) or os.listdir(self._data_dir) == []
+        return not os.path.exists(self._data_dir) or \
+                all(os.name != 'nt' and (n.startswith('.') or n == 'lost+found') for n in os.listdir(self._data_dir))
```
Comment (sdudoladov) [inline patroni/postgresql/__init__.py]: 1. Moving the `os.name != 'nt'` out of `all()` would make it slightly more readable . I imagine most readers expect only an iterable as an argument passed to `all`.
2. Windows apparently [also has hidden files](https://stackoverflow.com/a/14063074) but I guess it is out of scope

Comment (CyberDem0n) [reply patroni/postgresql/__init__.py]: 1. If you move `os.name != 'nt'` out of `all()` result would be wrong.
2. src/port/pgcheckdir.c: pg_check_dir() runs exaclty the same check.

File: tests/test_ha.py (hunk 2 above)
Comment (CyberDem0n) [reply tests/test_ha.py]: Well, actually it is related. Because it tests the [code](https://github.com/zalando/patroni/pull/1217/files#diff-57c53e988b775ee2902488fb33c1f433R1287-R1294) I was changing.

**Comment by sdudoladov** [comment]
:+1:

**Comment by CyberDem0n** [comment]
:+1:
